client/src/main/java/io/pravega/client/connection/impl/IoBuffer.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.connection.impl;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.pravega.common.MathHelpers;
+import io.pravega.shared.protocol.netty.WireCommands;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+class IoBuffer {
+    private int maxBufferSize = WireCommands.MAX_WIRECOMMAND_SIZE;
+    private ByteBuffer buffer = null;
+    
+    private ByteBuf sliceOut(int size) {
+        ByteBuf result = Unpooled.wrappedBuffer(buffer.array(), buffer.arrayOffset() + buffer.position(), size);
```

Comment:
Are we assuming that this is only a Heap ByteBuffer ?
`buffer.array()` won't work with a Direct Buffer

Reply:
The buffer is allocated locally in this class by wrapping an array. (And all accesses to it are private) So there is no way for it to be a heap byteBuffer.